packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,460 @@
+# Description
+
+Jobs is the Angular DevKit subsystem for scheduling and running generic functions with clearly 
+typed inputs and outputs. A Job is code (ie. a function) that can be scheduled, synchronized and
+schedule other jobs. Those Jobs can be run in parallel or follow an execution strategy. They can
+also be extended to allow for transformations, combination and refactoring through different 
+operations.
+
+The whole API is serializable, allowing someone to use a Node Stream or message channel to
+communicate between the job and the job scheduler.
+
+Jobs are lazy, cold and guaranteed to be only executed once per scheduling. Subscribing to a job
+will return events from the point where the job is at.
+
+## Argument, Input, Output and Channels
+Jobs receives an argument when first started. They can emit multiple outputs. In addition, jobs can
+listen to an input channel, and provide multiple output channels. Those channels emit asynchronous
+JSON messages that can be typed. In this context, input and output are always in relation to the 
+Job code itself, not the caller.
+
+If a Job was an executable, the argument would be the arguments on the command line, the input 
+channel would be STDIN, the output channel would be STDOUT, and the channels would be additional
```

Comment:
Does this mean that a job can be an executable? Or are you just comparing it to an executable?
Maybe something like "The I/O model is like that of an executable, where the argument corresponds to arguments on the command line, ..."

Reply:
Comparing to. Loved your phrasing, so used that. Thanks!